lib/web_ui/lib/src/engine/scene_view.dart

```diff
@@ -199,20 +235,14 @@ class EngineSceneView {
         sceneElement.insertBefore(container.container, currentElement);
       }
     }
-
-    // Remove any other unused containers
-    while (currentElement != null) {
-      final DomElement? sibling = currentElement.nextElementSibling;
-      sceneElement.removeChild(currentElement);
-      currentElement = sibling;
-    }
   }
 }
 
 sealed class SliceContainer {
   DomElement get container;
 
   void updateContents();
+  void dispose();
```

Comment:
A doc comment explaining who is responsible for calling this method would be helpful.